### PR TITLE
feat: show backup UI

### PR DESCRIPTION
**Pull request by rascanio97**

### CODE REVIEW

**Review by PixoDev — CHANGES_REQUESTED** [review]
Looking good, we may need to start thinking into adding tests to the project more consistently, as we are reliying a lot now in manual testing, and that doesn't scalate well

File: Shared/Services/BackupsService.swift [hunk 1]
```diff
@@ -111,10 +112,19 @@ class BackupsService: ObservableObject {
 
     func loadAllDevices() async {
         do {
-            self.deviceResponse = .success(try await DeviceService.shared.getAllDevices(deviceName: ConfigLoader().getDeviceName()))
+            let response: Result<[Device], Error> = .success(try await DeviceService.shared.getAllDevices(deviceName: ConfigLoader().getDeviceName()))
+            await MainActor.run { [weak self] in
```
[the remark this answers is not in the document]
Comment (rascanio97) [reply Shared/Services/BackupsService.swift:116]: bc it's raising a warning saying that we can't update a Published variable on a background thread. it needs to be in main thread and this is a way to do that in an `async` function. replacing the `DispatchQueue.main.async {}`

File: Shared/Services/BackupsService.swift [hunk 2]
```diff
@@ -136,20 +146,37 @@ class BackupsService: ObservableObject {
...
+            self?.hasOngoingBackup = false
+        }
     }
 
-    private func propagateSuccess(url: URL) {
-        logger.info("Backed up \(url) succesfully")
-        //TODO: propagate success backup to UI component
+    private func propagateSuccess() {
+        logger.info("Device backed up successfully")
+        DispatchQueue.main.async { [weak self] in
```
[the remark this answers is not in the document]
Comment (rascanio97) [reply Shared/Services/BackupsService.swift:171]: can't do this in the XPCService function callback. it throws an error

File: XPCBackupService/Entities/BackupTreeNode.swift [hunk 3]
```diff
@@ -68,24 +68,25 @@ class BackupTreeNode {
         return false
     }
     
-    func syncNodes() async throws -> Void {
+    func syncNodes(progress: Progress) async throws -> Void {
```
[the remark this answers is not in the document]
Comment (rascanio97) [reply XPCBackupService/Entities/BackupTreeNode.swift:71]: if i do this i will have a progress for each node, i want a progress for all nodes, and all trees.

[the remark this answers is not in the document]
Comment (PixoDev) [reply XPCBackupService/Entities/BackupTreeNode.swift]: You can still have a progress for each node, I believe that the progress you are sending here is the `backupProgress` in any case, you can send down the `backupProgress` here when generating the `BackupTreeNode`

https://github.com/internxt/drive-desktop-macos/blob/main/XPCBackupService/Services/BackupTreeGenerator.swift#L107

**Review by PixoDev — APPROVED** [review]

**Pull request merged**